src/engine/serverbrowser.cpp

```diff
@@ -273,32 +253,35 @@ void pingservers()
         enet_socket_set_option(pingsock, ENET_SOCKOPT_NONBLOCK, 1);
         enet_socket_set_option(pingsock, ENET_SOCKOPT_BROADCAST, 1);
     }
-    ENetBuffer buf;
+    int tmp_millis = totalmillis;
     uchar ping[MAXTRANS];
+    // ENet is crappy enough to invert it's members depending on
+    // target OS.
+    // This does also imply that this "efficient" library wastes 25%
+    // bytes in alignment on windows for this structure.
+#ifdef WIN32
+    ENetBuffer buf = { 0, ping };
+#else
+    ENetBuffer buf = { ping, 0 };
+#endif
     ucharbuf p(ping, sizeof(ping));
     putint(p, totalmillis ? totalmillis : 1);
+    buf.dataLength = p.length();
 
     static int lastping = 0;
-    if(lastping >= servers.length()) lastping = 0;
-    loopi(maxservpings ? min(servers.length(), maxservpings) : servers.length())
+    auto num_servers = servers.size();
```

Comment:
good point for const. For size and opt builds, I prefer to move repetitive calls into a variable. Makes debugging easier since you can inspect the variable, and shorter and thus more readable.